frontend/ts-src/lib/network.ts

```diff
@@ -0,0 +1,124 @@
+/* Corna frontend network utilities. */
+
+// axios types: https://github.com/axios/axios/blob/v1.x/index.d.ts
+import axios from "axios";
+import {
+    AxiosError,
+    AxiosHeaders,
+    AxiosPromise,
+    AxiosRequestConfig,
+    AxiosResponse,
+} from "axios";
+
+import { handlePromise } from "./utils.js";
+
+
+export function getApiUrl(): string {
+    /* Get the correct API URL.
+    *
+    * This is here to make local development easier.
+    * @returns { string }
+    */
+    const currUrl: string = window.location.hostname;
+    const apiUrl: string = (
+        (currUrl === "localhost" || currUrl === "127.0.0.1")
+        ? "http://127.0.0.1:8080/api"
+        : "https://api.mycorna.com"
+    )
+    return apiUrl;
+}
+
+
+export function request<T>(
+    urlExtension: string,
+    method: ("get" | "delete" | "post" | "put") = "get",
+    payload?: T,
+    headers?: AxiosHeaders,
+): AxiosPromise {
+    /* Generalised requestion function that wraps axios.request.
+    *
+    * @param { string } urlExtension: the API url extension
+    * @param { ("get" | "delete" | "post" | "put") } method:
+    *     http method for the request
+    * @param { optional<T> } payload: any payload to send
+    * @param { optional<AxiosHeaders> } headers: any headers to send
+    *     with request
+    * @returns { AxiosPromise }
+    */
+
+    const url: string = `${getApiUrl()}/${urlExtension}`;
+    const requestConf: AxiosRequestConfig = {
+        method: method,
```

Comment:
FYI you can omit a variable if the property name is spelled the same. For example, these are equivalent:
```
const hello = 'hi'

const x = {
  hello
}

const y = {
  hello: hello
}
```